typing_extensions/src_py3/typing_extensions.py

```diff
@@ -1595,3 +1604,93 @@ class Point2D(TypedDict):
     The class syntax is only supported in Python 3.6+, while two other
     syntax forms work for Python 2.7 and 3.2+
     """
+
+
+if HAVE_ANNOTATED:
+    class _Annotated(typing._GenericAlias, _root=True):
+        """Runtime representation of an annotated type.
+
+        At its core 'Annoted[t, dec1, dec2...]' is an alias for the type 't'
+        with extra annotations. The alias behaves like a normal typing alias,
+        instantiating is the same as instantiating the underlying type, binding
+        it to types is also the same...
+        """
+
+        def __init__(self, origin, extras):
+            if isinstance(origin, _Annotated):
+                extras = origin.__extras__ + extras
+                origin = origin.__origin__
+            super().__init__(origin, origin)
+            self.__extras__ = extras
```

Comment:
I would not call this `__extras__` to avoid confusion with an attribute with similar name in older versions of `typing`. Maybe call it `__metadata__`?